lychee-lib/src/client.rs

```diff
@@ -1076,4 +1112,32 @@ mod tests {
             assert!(res.status().is_unsupported());
         }
     }
+
+    #[tokio::test]
+    async fn test_chain() {
+        use reqwest::Request;
+
+        #[derive(Debug)]
+        struct ExampleHandler();
+
+        impl Chainable<Request, Status> for ExampleHandler {
+            fn handle(&mut self, _: Request) -> ChainResult<Request, Status> {
+                ChainResult::EarlyExit(Status::Excluded)
+            }
+        }
+
+        let chain = Arc::new(Mutex::new(RequestChain::new(vec![Box::new(
+            ExampleHandler {},
+        )])));
```

Comment:
With some refactoring, we might be able to make this

```rust
let chain = RequestChain::new([ExampleHandler {}]);
```

of 

```rust
let chain = RequestChain::from([ExampleHandler {}]);
```

This can be achieved by
- Making `Arc<Mutex<...>>` part of the `RequestChain`. It's an implementation detail.
- Making `RequestChain::new` accept a `[impl Chainable<_,_>]` 
- Boxing each Handler inside `new`.

Reply:
Will make `Arc<Mutex<...>` part of `RequestChain` :+1: 

But didn't succeed in passing an array like this. I need to pass it as reference, because `the size for values of type `[impl Chainable<T, R> + Send]` cannot be known at compilation time`. Passing as reference then means that the elements itself also need to be references which would complicate matters a lot, I think.

```
expected struct `std::vec::Vec<std::boxed::Box<(dyn chain::Chainable<T, R> + std::marker::Send + 'static)>>`
   found struct `std::vec::Vec<std::boxed::Box<&impl Chainable<T, R> + Send>>` [E0308]
```